ironfish/src/rpc/routes/wallet/multisig/dkg/round1.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { multisig } from '@ironfish/rust-nodejs'
+import * as yup from 'yup'
+import { RPC_ERROR_CODES, RpcValidationError } from '../../../../adapters'
+import { ApiNamespace } from '../../../namespaces'
+import { routes } from '../../../router'
+import { AssertHasRpcContext } from '../../../rpcContext'
+
+export type DkgRound1Request = {
+  // TODO `secretName` or `multisigSecretName`?
+  name: string
+  minSigners: number
+  participants: Array<{ identity: string }>
+}
+
+export type DkgRound1Response = {
+  encryptedSecretPackage: string
+  publicPackage: string
+}
+
+export const DkgRound1RequestSchema: yup.ObjectSchema<DkgRound1Request> = yup
+  .object({
+    name: yup.string().defined(),
+    minSigners: yup.number().defined(),
+    participants: yup
+      .array()
+      .of(yup.object({ identity: yup.string().defined() }).defined())
+      .defined(),
+  })
+  .defined()
+
+export const DkgRound1ResponseSchema: yup.ObjectSchema<DkgRound1Response> = yup
+  .object({
+    encryptedSecretPackage: yup.string().defined(),
+    publicPackage: yup.string().defined(),
+  })
+  .defined()
+
+routes.register<typeof DkgRound1RequestSchema, DkgRound1Response>(
+  `${ApiNamespace.wallet}/multisig/dkg/round1`,
+  DkgRound1RequestSchema,
+  async (request, node): Promise<void> => {
+    AssertHasRpcContext(request, node, 'wallet')
+
+    const { name, minSigners, participants } = request.data
+
+    // TODO check if this secret is already being used for an account (there's
+    // nothing inherently wrong with using the same secret for multiple
+    // multisig groups, but I think our code assumes a 1:1 relationship between
+    // secret names and account names)
```

Comment:
I don't think we have a way to prevent this at all with TDK 🤔

I think we should avoid relying on a 1:1 relationship between secret names and account names

Reply:
Good to know, thanks. I must have misread some code I found earlier.